File: springapp/src/main/java/com/examly/springapp/service/DeviceService.java
package com.examly.springapp.service;


import net.java.SpringbootBackend.exception.ResourceNotFoundException;
import net.java.SpringbootBackend.model.Device;
import net.java.SpringbootBackend.repository.DeviceRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class DeviceService {
        private final DeviceRepository deviceRepository;

        @Autowired
        public DeviceService(DeviceRepository deviceRepository) {
                this.deviceRepository = deviceRepository;
        }

        public List<Device> getAllDevices() {
                return deviceRepository.findAll();
        }

        public Device getDeviceById(Long deviceId) {
                return deviceRepository.findById(deviceId)
                        .orElseThrow(() -> new ResourceNotFoundException("Device not exist with id:"+ deviceId));
        }

        public Device createDevice(Device device) {
                return deviceRepository.save(device);
        }

        public Device updateDevice(Long deviceId, Device deviceDetails) {
                Device device = getDeviceById(deviceId);
                device.setBrand(deviceDetails.getBrand());
                device.setType(deviceDetails.getType());
                device.setModel(deviceDetails.getModel());

                // Update other fields as needed
                return deviceRepository.save(device);
        }

        public void deleteDevice(Long deviceId) {
                deviceRepository.deleteById(deviceId);
        }
}

File: reactapp/src/services/DeviceService.js
import axios from "axios";
const DEVICE_BASE_REST_API_URL ="https://8080-fabdbbbdadfdcdabfcdcbbdbeaabbeaeaadbdbabf.project.examly.io/devices";

class DeviceService{
    getAllDevices(){
        return axios.get(DEVICE_BASE_REST_API_URL)
    }
    createDevice(device){
        return axios.post(DEVICE_BASE_REST_API_URL, device)
    }
    getDeviceById(deviceId){
         return axios.get(DEVICE_BASE_REST_API_URL + '/' + deviceId);
    }
    updateDevice(deviceId,device){
        return axios.put(DEVICE_BASE_REST_API_URL +'/' +deviceId, device);
    }
    deleteDevice(deviceId){
        return axios.delete(DEVICE_BASE_REST_API_URL +'/' +deviceId);
    }

    
}
// eslint-disable-next-line
export default new DeviceService();